File: 787/src.cpp
#include <bits/stdc++.h>

using namespace std;

using ll = long long int;

void compute(vector<long long int> &p)
{
    if(p.size()==0)
    {
        std::cout<<0<<std::endl;
        return;
    }
    long long int mx = p[0];
    vector<ll> dp = {p[0]};
    for(ll i=1;i<p.size();i++)
    {
        dp.push_back(dp[i-1]*p[i]);
    }
    for(ll i=0;i<p.size();i++)
    {
        for(ll j=0;j<i;j++)
        {
            if(dp[j]==0)
                continue;
            auto x = dp[i]/dp[j];
            mx = max(mx,x);
        }
    }
    std::cout<<mx<<std::endl;
}

int main()
{
	long long int n;
	vector<long long int> p;
	while(cin>>n)
	{
		if(n==-999999)
		{
			compute(p);
			p={};
			continue;
		}
        p.push_back(n);
	}
	return 0;
}

File: 787/src.py
#!/usr/bin/python3

def compute(p):
    if len(p)==0:
        print(0)
        return
    mx = p[0]
    dp = [p[0]]
    for i in range(1,len(p)):
        dp.append(dp[i-1]*p[i])
    for i in range(len(p)):
        for j in range(i):
            if dp[j]==0:
                continue
            x = dp[i]/dp[j]
            mx = max(mx,x)

    print(int(mx))

p = []
while True:
  try:
    text = input()
    nums = text.split(" ")
    for x in nums:
        x = int(x)
        if x==-999999:
            compute(p)
            p = []
            continue
        p.append(x)

  except EOFError:
    break
